File: src/com/jetbrains/idear/WordToNumberConverter.java
package com.jetbrains.idear;

/**
 * Created by breandan on 10/25/2015.
 */
public class WordToNumberConverter {
    public static int getNumber(String input) {
        int accumulator = 0;
        int total = 0;
        String[] words = input.trim().split("\\s+");

        for (String word : words) {
            switch (word) {
                case "zero":
                    accumulator += 0;
                    break;
                case "one":
                    accumulator += 1;
                    break;
                case "two":
                    accumulator += 2;
                    break;
                case "three":
                    accumulator += 3;
                    break;
                case "four":
                    accumulator += 4;
                    break;
                case "five":
                    accumulator += 5;
                    break;
                case "six":
                    accumulator += 6;
                    break;
                case "seven":
                    accumulator += 7;
                    break;
                case "eight":
                    accumulator += 8;
                    break;
                case "nine":
                    accumulator += 9;
                    break;
                case "ten":
                    accumulator += 10;
                    break;
                case "eleven":
                    accumulator += 11;
                    break;
                case "twelve":
                    accumulator += 12;
                    break;
                case "thirteen":
                    accumulator += 13;
                    break;
                case "fourteen":
                    accumulator += 14;
                    break;
                case "fifteen":
                    accumulator += 15;
                    break;
                case "sixteen":
                    accumulator += 16;
                    break;
                case "seventeen":
                    accumulator += 17;
                    break;
                case "eighteen":
                    accumulator += 18;
                    break;
                case "nineteen":
                    accumulator += 19;
                    break;
                case "twenty":
                    accumulator += 20;
                    break;
                case "thirty":
                    accumulator += 30;
                    break;
                case "forty":
                    accumulator += 40;
                    break;
                case "fifty":
                    accumulator += 50;
                    break;
                case "sixty":
                    accumulator += 60;
                    break;
                case "seventy":
                    accumulator += 70;
                    break;
                case "eighty":
                    accumulator += 80;
                    break;
                case "ninety":
                    accumulator += 90;
                    break;
                case "hundred":
                    accumulator *= 100;
                    break;
                case "thousand":
                    accumulator *= 1000;
                    total += accumulator;
                    accumulator = 0;
                    break;
            }
        }

        return total + accumulator;
    }
}

File: src/main/java/org/openasr/idear/WordToNumberConverter.kt
package com.jetbrains.idear

/**
 * Created by breandan on 10/25/2015.
 */
object WordToNumberConverter {
    fun getNumber(input: String): Int {
        var accumulator = 0
        var total = 0
        val words = input.trim { it <= ' ' }.split("\\s+".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()

        for (word in words) {
            when (word) {
                "zero" -> accumulator += 0
                "one" -> accumulator += 1
                "two" -> accumulator += 2
                "three" -> accumulator += 3
                "four" -> accumulator += 4
                "five" -> accumulator += 5
                "six" -> accumulator += 6
                "seven" -> accumulator += 7
                "eight" -> accumulator += 8
                "nine" -> accumulator += 9
                "ten" -> accumulator += 10
                "eleven" -> accumulator += 11
                "twelve" -> accumulator += 12
                "thirteen" -> accumulator += 13
                "fourteen" -> accumulator += 14
                "fifteen" -> accumulator += 15
                "sixteen" -> accumulator += 16
                "seventeen" -> accumulator += 17
                "eighteen" -> accumulator += 18
                "nineteen" -> accumulator += 19
                "twenty" -> accumulator += 20
                "thirty" -> accumulator += 30
                "forty" -> accumulator += 40
                "fifty" -> accumulator += 50
                "sixty" -> accumulator += 60
                "seventy" -> accumulator += 70
                "eighty" -> accumulator += 80
                "ninety" -> accumulator += 90
                "hundred" -> accumulator *= 100
                "thousand" -> {
                    accumulator *= 1000
                    total += accumulator
                    accumulator = 0
                }
            }
        }

        return total + accumulator
    }
}